src/qa4sm_reader/plot_utils.py

```diff
@@ -67,18 +86,27 @@ def geotraj_to_geo2d(df, var, index=globals.index_names):
     yy = df.index.get_level_values(index[0])   # lat
     data = df[var]
 
-    x_min, x_max, dx, len_x = _get_grid(xx)
-    y_min, y_max, dy, len_y = _get_grid(yy)
+    if grid_stepsize not in ['nan', None]:
+        x_min, x_max, dx, len_x = _get_grid_for_irregulars(xx, grid_stepsize)
+        y_min, y_max, dy, len_y = _get_grid_for_irregulars(yy, grid_stepsize)
+
+        data_extent = (x_min - dx, x_max + dx, y_min - dy, y_max + dy)
+        zz, grid = oversample(xx, yy, data.values, data_extent, dx, dy)
```

Comment:
Where is this function defined? If I give it a netcdf file with a stepsize that is not nan, it fails